tei_adapter/terraform/topics.tf

```diff
@@ -15,3 +19,15 @@ data "aws_iam_policy_document" "publish_to_tei_id_extractor_topic" {
     ]
   }
 }
+
+data "aws_iam_policy_document" "publish_to_tei_adapter_topic" {
+  statement {
+    actions = [
+      "sns:Publish",
+    ]
+
+    resources = [
+      module.tei_adapter_topic.arn
+    ]
+  }
+}
```

Comment:
There's a `publish_policy` output on our SNS topic module you can use rather than declaring it again.